Makefile

```diff
@@ -11,35 +13,27 @@ REMOTEDIR=deb.qcode.co.uk
 
 .PHONY: all test
 
-all: check-version test package upload clean
+all: check-version package test upload clean
 package: check-version
 	# Copy files to pristine temporary directory
+	rm -rf $(TEMP_PATH)	
+	mkdir $(TEMP_PATH)
 	rm -rf package
 	mkdir package
 	curl --fail -K ~/.curlrc_github -L -o v$(VERSION).tar.gz https://api.github.com/repos/qcode-software/qcode-tcl/tarball/v$(VERSION)
-	tar --strip-components=1 --exclude Makefile --exclude description-pak --exclude doc --exclude docs.tcl --exclude package.tcl --exclude test --exclude test_all.tcl -xzvf v$(VERSION).tar.gz -C package
-	./package.tcl tcl package ${NAME} ${VERSION}
+	tar --strip-components=1 --exclude Makefile --exclude description-pak --exclude doc --exclude docs.tcl --exclude package.tcl -xzvf v$(VERSION).tar.gz -C $(TEMP_PATH)
+	./package.tcl $(TEMP_PATH)/tcl package ${NAME} ${VERSION}
 	./pkg_mkIndex package
 	# checkinstall
-	fakeroot checkinstall -D --deldoc --backup=no --install=no --pkgname=$(DPKG_NAME) --pkgversion=$(VERSION) --pkgrelease=$(RELEASE) -A all -y --maintainer $(MAINTAINER) --pkglicense="BSD" --reset-uids=yes --requires "tcl,tcllib,html2text,curl,tclcurl" --replaces none --conflicts none make local-install
+	fakeroot checkinstall -D --deldoc --backup=no --install=no --pkgname=$(DPKG_NAME) --pkgversion=$(VERSION) --pkgrelease=$(RELEASE) -A all -y --maintainer $(MAINTAINER) --pkglicense="BSD" --reset-uids=yes --requires "tcl,tcllib,html2text,curl,tclcurl" --replaces none --conflicts none make install
 
-tcl-package : check-version
-	rm -rf package
-	mkdir package
-	./package.tcl tcl package ${NAME} ${VERSION}
-	./pkg_mkIndex package
-
-test: tcl-package 
-	tclsh ./test_all.tcl -testdir test
-	rm -rf package
-
-install: tcl-package local-install
+test: package
+	cd $(TEMP_PATH)/test && tclsh all.tcl
 
-local-install:
+install:  
 	mkdir -p /usr/lib/tcltk/$(NAME)$(VERSION)
 	cp package/*.tcl /usr/lib/tcltk/$(NAME)$(VERSION)/
```

Comment:
Will the package directory exists without running `tcl-package` first?

Reply:
No it will not exist.